skyportal/tests/api/test_comments_on_spectrum.py

```diff
@@ -0,0 +1,249 @@
+from skyportal.tests import api
+import datetime
+
+
+def test_add_and_retrieve_comment_group_id(
+    comment_token, upload_data_token, public_source, public_group, lris
+):
+    status, data = api(
+        'POST',
+        'spectrum',
+        data={
+            'obj_id': str(public_source.id),
+            'observed_at': str(datetime.datetime.now()),
+            'instrument_id': lris.id,
+            'wavelengths': [664, 665, 666],
+            'fluxes': [234.2, 232.1, 235.3],
+        },
+        token=upload_data_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+    spectrum_id = data["data"]["id"]
+
+    status, data = api(
+        'POST',
+        'comment',
+        data={
+            'obj_id': public_source.id,
+            'spectrum_id': spectrum_id,
+            'text': 'Comment text',
+            'group_ids': [public_group.id],
+        },
+        token=comment_token,
+    )
+    assert status == 200
+    comment_id = data['data']['comment_id']
+
+    status, data = api('GET', f'comment/{comment_id}/spectrum', token=comment_token)
+
+    assert status == 200
+    assert data['data']['text'] == 'Comment text'
+
+
+def test_add_and_retrieve_comment_group_access(
```

Comment:
Awesome, thanks for making these tests thoroughly cover the access rules as well!